scripts/controllers/usefulness_controller.ts

```diff
@@ -0,0 +1,58 @@
+import { Controller } from '@hotwired/stimulus'
+
+class UsefulnessController extends Controller {
+    vote: String | null
+    launchModal: String | null
+    
+    initialize() {
+      this.vote = null
+      this.launchModal = ''
+    }
+  
+    static get targets() {
+      return []
+    }
+  
+    handleVote(event: any) {
+      this.launchModal = event.params.launchModal
+      this.vote = this.launchModal === 'useful-yes' ? 'yes' : 'no'
+      const modal = document.querySelector(`#${this.launchModal}`)
+      modal.classList.add('active')
+    }
+  
+    handleCancel() {
+      this.closeModal()
+    }
+  
+    handleSubmit() {
+      const documentId = this.data.get('documentId')
+      const location = this.vote === 'yes' ? '/api/upvote' : '/api/downvote'
+      const checked: HTMLInputElement = document.querySelector('input[name="reason"]:checked')
```

Comment:
can you use `targets` for this?

https://stimulus.hotwired.dev/reference/targets#shared-targets has a checkbox example

Reply:
Yep!  Done.